### PR TITLE
Update simple_paths.py to improve readability of the BFS.

### PR DESCRIPTION
**Pull request by SultanOrazbayev**
Improve readability of the BFS.

Resolves https://github.com/networkx/networkx/issues/3912.

### CODE REVIEW

File: networkx/algorithms/simple_paths.py [hunk 1]
```diff
@@ -257,7 +257,7 @@ def _empty_generator():
 
 
 def _all_simple_paths_graph(G, source, targets, cutoff):
-    visited = dict.fromkeys([source])
+    visited = dict.fromkeys([source], True)
```
Comment (dschult) [inline networkx/algorithms/simple_paths.py:260]: Would this be more readable? 
```suggestion
    visited = {source: True}
```

**Comment by SultanOrazbayev** [comment]
Agree with the change, I hesitated with proposing it initially, but agree that it's more readable that way.

**Review by dschult — APPROVED** [review]
Thanks!   LGTM

**Review by MridulS — APPROVED** [review]
Thanks @SultanOrazbayev!

**Pull request merged**